### DISCUSSION TITLE
Start Parameter or Profile to choose Color Profile and Look and Feel

### DISCUSSION THREAD

**Discussion by AJK-Muc** [apache/netbeans#7596] [Q&A]
I wish NetBeans has support for automatic dark mode from the OS settings, or at least a function to change the color profiles with one click instead of 11 (see bullet list). Since this is not possible,  see #5540, I have to perform this 11 clicks sequence at least twice a day (sunset and sunrise) or more often depending in what environment I have to work:

1. Menu
2. Settings
3. Font & Colors
4. Profile
5. Light / Dark
6. Apply
7. Appearance
8. Look and Feel
9. Light / Dark
10. OK
11. Restart

Does anyone know if there is any way to prevent those 11 clicks and set these options in a start script with start parameters or to different profiles choosing from command line? So I could only have to quit the program and start it again with the other script or profile?

**Comment by neilcsmith-net** [comment]
There's a one click button on the dashboard that should do exactly this for you.

**Reply by AJK-Muc** [reply]
Yes, I know, if that function would allow custom settings it would be great. If there would be a tutorial available, I guess it would be easy to figure out which start parameter to set. [But since "NetBeans Wiki Index" has no content](https://netbeans.apache.org/wiki/main/wiki/), I don't know where to find some information about that.

**Reply by AJK-Muc** [reply]
It would be much easier to change just the preferred theme with a start parameter or profile (like the switch in dashboard), than to edit all needed parameters in 
~/Library/Application Support/NetBeans/22/config/LookAndFeel/FlatLaf.properties and
~/Library/Application Support/NetBeans/22/config/Editors/FontsColors/{Theme}/*.xml

**Reply by mbien** [reply]
this is so that you can create two shortcuts, one for dark mode one for light mode?

**Reply by AJK-Muc** [reply]
> There's a one click button

Could you please tell me in which file the results of that button switch gets written or what's the syntax to include into a file like `NetBeans/{Version}/config/.nbattrs` so that I could change it on demand?  Like `<attr name="currentFontColorProfile" stringvalue="FlatLaf Dark"/>` I wish there was an easy to find reference to find the right options to select the same theme in LookAndFeel.

I could simply use the current systems darkmode setting in a shell script and use `sed` to change some values to "Light" or "Dark" accordingly when opening NetBeans. Calling `netbeans --laf = 'FlatLaf Dark'` does not work, it's obviously not the correct name but I cant't find anywhere the names to use for the available themes 'FlatLaf Light' and 'FlatLaf Dark' used in the one click button.
